### PR TITLE
[WIP] lbryio internal API calls and rewards component

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: src/lbryio.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            .then(() => true)
+            .catch(() => false);
+        })
+        .then(isTokenValid => {
+          if (isTokenValid) {
+            return reject;
+          }
+
+          return Lbry.status().then(status => {
+            const store = global.store || (window.app ? window.app.store : null);
```
Comment (seanyesmunt) [inline src/lbryio.js]: Do you need to check that `window` is an object so it doesn't break android?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export { Lbryio } from 'lbryio';
```
Comment (seanyesmunt) [inline src/index.js]: Might as well export the redux stuff too.

File: src/redux/actions/auth.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import * as ACTIONS from 'constants/action_types';
+import Lbryio from 'lbryio';
+
+export function doNewInstallation(installationId) {
```
Comment (kauffj) [inline src/redux/actions/auth.js]: Isn't this more like `doAuthentication` then `doNewInstallation`?

File: src/constants/action_types.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const GET_AUTH_TOKEN_STARTED = 'GET_AUTH_TOKEN_STARTED';
```
Comment (kauffj) [inline src/constants/action_types.js]: This should probably be `AUTHENTICATION_START`, `AUTHENTICATION_SUCCESS`, and `AUTHENTICATION_FAIL`. (`AUTH_XXX` would also be acceptable).

https://github.com/lbryio/lbry-app/blob/171ea2b97b6a393c880995a42cd4189cdda3e5bd/src/renderer/constants/action_types.js#L2

File: src/redux/actions/auth.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        }
+      })
+      .catch(() => {
+        dispatch({
+          type: ACTIONS.GET_AUTH_TOKEN_COMPLETED,
+          data: { authToken: null },
+        });
+      });
+  };
+}
```
Comment (kauffj) [inline src/redux/actions/auth.js]: Guessing you just haven't gotten to it, but worth porting this too https://github.com/lbryio/lbry-app/blob/171ea2b97b6a393c880995a42cd4189cdda3e5bd/src/renderer/redux/actions/user.js#L39

File: src/lbryio.js (hunk 1 above)
Comment (kauffj) [reply src/lbryio.js]: @seanyesmunt @akinwale if `window` isn't safe to use on Android then I'd propose the convention that it's unacceptable to use it in all shared JS code and instead must be passed in as a dependency. Littering code with `window && window.foo()` will hurt maintainability, readability, and introduce additional sources of error.

Comment (kauffj) [reply src/lbryio.js]: To be clear, I'm not suggesting passing in `window`, specifically, but instead a generic term (such as `store` or whatever is appropriate) to be used in it's stead.

Comment (akinwale) [reply src/lbryio.js]: Looks like `window` exists is actually already set to `global` React Native (https://github.com/facebook/react-native/blob/master/Libraries/Core/InitializeCore.js#L34-L36), so I can remove this.

**Comment by akinwale** [comment]
@kauffj @seanyesmunt Pushed a new commit which includes the rewards actions/reducers/selectors, user reducers and selectors and a subset of user actions.

**Comment by skhameneh** [comment]
Looks good to me, I'll wait for @seanyesmunt's input on this

**Pull request merged**